src/app/material.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+import { environment } from './../environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MaterialService {
+
+  private _url: string = `${environment.apiUrl}/materials/`;
+  constructor(private http: HttpClient) { }
+
+  getMaterials(): Observable<any> {
+    return this.http.get(this._url);
+  }
+
+  getMaterialById(id): Observable<any> {
+    return this.http.get(this._url + id + '/');
+  }
+
+  updateMaterial(material): Observable<any> {
+    const body = {
+      title: material.title,
```

Comment:
You can simplify removing a property with object destructuring in Typescript / ES6.
`const { material_id, ...body } = material;`